phase4-entsog-client/src/main/java/com/helger/phase4/entsog/Phase4ENTSOGSender.java

```diff
@@ -104,36 +103,35 @@ public final IMPLTYPE setEncryptionKeyIdentifierType(ECryptoKeyIdentifierType en
 		public final IMPLTYPE setSigningKeyIdentifierType(ECryptoKeyIdentifierType signingKeyIdentifierType) {
 			m_signingKeyIdentifierType = signingKeyIdentifierType;
 			return thisAsT();
-		}		
-		
+		}
+
 		/**
 		 * Set the payload to be send out.
 		 *
 		 * @param aBuilder The payload builder to be used.
 		 * @return this for chaining
 		 */
 		@Nonnull
-		public final IMPLTYPE payload(@Nullable final Phase4OutgoingAttachment.Builder aBuilder, Charset aPayloadCharset, String sPayloadDocumentType) {
+		public final IMPLTYPE payload(@Nullable final Phase4OutgoingAttachment.Builder aBuilder,
+				@Nullable ENTSOGPayloadParams aPayloadParams) {
```

Comment:
Just wrapped params together... I'll ask you to to move charset to phase4 core. I don't know if you want to use charset inside for any reason.

Reply:
The charset is already inside the current master branch - see https://github.com/phax/phase4/commit/9f8ed8d9804c3b46515a1df4ba25a7f85201791f